### DISCUSSION TITLE
Magmacore on Apple M1 or M2

### DISCUSSION THREAD

**Discussion by salvatoreloreto** [magma/magma#14854] [Q&A]
Hi there

I would like to start to get familiar with magma, so I am wondering if it is possible to get it Run on a MacBook with a M1 or M2 processor!
Is there anything that has to be documented about How to do, if there anything that is different from the installation on x86

thanks for the help
/Sal

**Comment by salvatoreloreto** [comment]
I am trying to install magma on my MacBook Pro with an M2 processor
when I try to start up magma I got the following error

MBP gateway % vagrant up magma
Bringing machine 'magma' up with 'virtualbox' provider...
==> magma: You assigned a static IP ending in ".1" to this machine.
==> magma: This is very often used by the router and can cause the
==> magma: network to not work properly. If the network doesn't work
==> magma: properly, try changing this IP.
==> magma: Cloning VM...
==> magma: Matching MAC address for NAT networking...
==> magma: You assigned a static IP ending in ".1" to this machine.
==> magma: This is very often used by the router and can cause the
==> magma: network to not work properly. If the network doesn't work
==> magma: properly, try changing this IP.
==> magma: Checking if box 'magmacore/magma_dev' version '1.1.20210618-patched' is up to date...
==> magma: Setting the name of the VM: magma-dev
==> magma: Clearing any previously set network interfaces...
==> magma: Destroying VM and associated drives...
Traceback (most recent call last):
	135: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/batch_action.rb:86:in `block (2 levels) in run'
	134: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/machine.rb:203:in `action'
	133: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/machine.rb:203:in `call'
	132: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/environment.rb:631:in `lock'
	131: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/machine.rb:217:in `block in action'
	130: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/machine.rb:248:in `action_raw'
	129: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `run'
	128: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/busy.rb:19:in `busy'
	127: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `block in run'
	126: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builder.rb:180:in `call'
	125: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	124: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/check_virtualbox.rb:26:in `call'
	123: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	122: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/call.rb:53:in `call'
	121: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `run'
	120: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/busy.rb:19:in `busy'
	119: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `block in run'
	118: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builder.rb:180:in `call'
	117: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	116: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	115: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	114: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
	113: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	112: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	111: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	110: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	109: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	108: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/call.rb:53:in `call'
	107: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `run'
	106: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/busy.rb:19:in `busy'
	105: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `block in run'
	104: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builder.rb:180:in `call'
	103: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	102: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	101: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	100: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/check_accessible.rb:18:in `call'
	 99: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 98: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/customize.rb:40:in `call'
	 97: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 96: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/import_master.rb:36:in `call'
	 95: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 94: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/prepare_clone.rb:15:in `call'
	 93: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 92: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/prepare_clone_snapshot.rb:32:in `call'
	 91: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 90: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/import.rb:11:in `call'
	 89: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/import.rb:46:in `clone'
	 88: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 87: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/discard_state.rb:15:in `call'
	 86: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 85: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/match_mac_address.rb:22:in `call'
	 84: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 83: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	 82: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 81: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/env_set.rb:19:in `call'
	 80: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 79: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/check_virtualbox.rb:26:in `call'
	 78: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 77: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	 76: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 75: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/box_check_outdated.rb:93:in `call'
	 74: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 73: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/call.rb:53:in `call'
	 72: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `run'
	 71: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/busy.rb:19:in `busy'
	 70: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `block in run'
	 69: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builder.rb:180:in `call'
	 68: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 67: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	 66: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 65: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/call.rb:53:in `call'
	 64: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `run'
	 63: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/busy.rb:19:in `busy'
	 62: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `block in run'
	 61: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builder.rb:180:in `call'
	 60: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 59: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	 58: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 57: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/call.rb:53:in `call'
	 56: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `run'
	 55: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/busy.rb:19:in `busy'
	 54: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/runner.rb:101:in `block in run'
	 53: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builder.rb:180:in `call'
	 52: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 51: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	 50: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 49: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/check_accessible.rb:18:in `call'
	 48: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 47: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/clean_machine_folder.rb:26:in `call'
	 46: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 45: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/set_name.rb:50:in `call'
	 44: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 43: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/clear_forwarded_ports.rb:15:in `call'
	 42: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 41: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/provision.rb:80:in `call'
	 40: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 39: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/env_set.rb:19:in `call'
	 38: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 37: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/prepare_forwarded_port_collision_params.rb:30:in `call'
	 36: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 35: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/handle_forwarded_port_collisions.rb:52:in `call'
	 34: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 33: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/prepare_nfs_valid_ids.rb:12:in `call'
	 32: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 31: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/synced_folders/nfs/action_cleanup.rb:25:in `call'
	 30: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 29: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/synced_folder_cleanup.rb:28:in `call'
	 28: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 27: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/delayed.rb:19:in `call'
	 26: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 25: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/builtin/synced_folders.rb:87:in `call'
	 24: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 23: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/prepare_nfs_settings.rb:19:in `call'
	 22: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 21: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/set_default_nic_type.rb:64:in `call'
	 20: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 19: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/clear_network_interfaces.rb:26:in `call'
	 18: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/action/warden.rb:48:in `call'
	 17: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/network.rb:109:in `call'
	 16: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/network.rb:109:in `each'
	 15: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/network.rb:121:in `block in call'
	 14: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/network.rb:381:in `hostonly_adapter'
	 13: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/action/network.rb:499:in `hostonly_create_network'
	 12: from /opt/vagrant/embedded/lib/ruby/2.7.0/forwardable.rb:235:in `create_host_only_network'
	 11: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/driver/version_7_0.rb:125:in `create_host_only_network'
	 10: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/driver/base.rb:398:in `execute'
	  9: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/retryable.rb:17:in `retryable'
	  8: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/driver/base.rb:403:in `block in execute'
	  7: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/driver/base.rb:467:in `raw'
	  6: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/busy.rb:19:in `busy'
	  5: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/plugins/providers/virtualbox/driver/base.rb:468:in `block in raw'
	  4: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/subprocess.rb:22:in `execute'
	  3: from /opt/vagrant/embedded/gems/2.3.4/gems/vagrant-2.3.4/lib/vagrant/util/subprocess.rb:84:in `execute'
	  2: from /opt/vagrant/embedded/gems/2.3.4/gems/childprocess-4.1.0/lib/childprocess.rb:23:in `new'
	  1: from /opt/vagrant/embedded/gems/2.3.4/gems/childprocess-4.1.0/lib/childprocess.rb:23:in `new'
/opt/vagrant/embedded/gems/2.3.4/gems/childprocess-4.1.0/lib/childprocess/abstract_process.rb:44:in `initialize': all arguments must be String: ["/usr/local/bin/VBoxManage", "hostonlynet", "add", "--name", "vagrantnet-vbox3", "--netmask", 64, "--lower-ip", "3001::", "--upper-ip", "3001::"] (ArgumentError)

**Reply by sineer** [reply]
This error has nothing todo with ARM / m1/m2/m3 mac.

I'm trying to setup magma 1.8 on INTEL mac (Monterey mac mini 2014) and had exact same error from VirtualBox!

So I downgraded from their latest 7.0.x to 6.1.50 and got passed that error!

Meanwhile, I still can't get bare-metal AGW working on latest x64 20.04 Ubuntu LTS even after patching a few active bugs...

**Comment by lucasgonze** [comment]
Short answer: if you have an Intel machine around, you'll get rolling a lot faster.

I have generally been able to address pointpoint issues using Docker on my M1 Mac, but most dev is faster on my Intel Mac.

**Reply by salvatoreloreto** [reply]
Thanks for the answer.
do you know if the new Magma version is planning to support Apple silicon (i.e., M1/M2) ?

Would be possible for you to share what you have done to Docker on M1 MAC?

BR
Sal

**Reply by lucasgonze** [reply]
I wasn't able to get the Vagrantfile working. What did work for me was fiddling around with Docker on a component-by-component basis. Like, this is my notes file for working with documentation:

- start docker desktop
- cd docs; make dev
- open localhost:3000

Some other items from my notes that might help you find a way in:

- magma/orc8r/cloud/docker/build.py is for building the orc8r, docker-compose build magmalte is for building the NMS. 
- nms/README.md has lots of useful background info. It says that e2e_test_setup.sh brings up the NMS docker containers, as well as a mock Orc8r API server.

**Reply by salvatoreloreto** [reply]
thanks Lucas

what to build and run the AGW? 
at moment I am interested to only est/develop the LTE features of AGW, without cloud based network management

**Reply by lucasgonze** [reply]
I would have to poke around too, my friend. I realize that's easier for me than a person new to the project - sorry.  In my experience, there is usually a README or Dockerfile in some corner of the code that will lead in the right direction.

You might also try your question in Slack. The community is a lot more active there. The best channel is probably #project-arm-support.

**Reply by lucasgonze** [reply]
This might be a useful stepping stone: https://github.com/canonical/charmed-magma

**Comment by lucasgonze** [comment]
Bruno says that he personally gets by on an M1 by not using Vagrant/Virtualbox. You have to open up those scripts and get the CLI commands going in your personal shell. He uses Qemu.

If there are people using M1 who want to share tips and tricks, this thread would be a good place.
